cluster-autoscaler/processors/nodegroupset/balancing_processor.go

```diff
@@ -46,6 +46,18 @@ func (b *BalancingNodeGroupSetProcessor) FindSimilarNodeGroups(context *context.
 			"failed to find template node for node group %s",
 			nodeGroupId)
 	}
+
+	ignoredLabels := BasicIgnoredLabels
```

Comment:
Intiialize this to empty map and always copy context.IgnoredLabels  and BasicIgnoredLabels.
We do not want to modify global variable as we do now by setting cp-specific values in it to true.